### DISCUSSION TITLE
Pass environment variable to spawn

### DISCUSSION THREAD

**Discussion by addy419** [qtile/qtile#2794] [Q&A]
I want run `LD_PRELOAD=/usr/lib/libwcwidth-icons.so urxvt` with lazy.spawn but cannot find a way to set environment variables for a single spawn (I don't want to set it globally)

**Comment by Fabien-Bernard** [comment]
I don't know if there is a more "qtile" way to do it, but you could put your commands in a shell script and call this script via the lazy.spawn.

**Comment by h-mnzlr** [comment]
I reckon you could just seperaten them with a semicolon and put it on one line. Haven't tried it though.

**Comment by bodograumann** [comment]
The syntax you show needs a shell. You can wrap your command in a shell like this:
```python
lazy.spawn("sh -c 'LD_PRELOAD=/usr/lib/libwcwidth-icons.so urxvt'")
```

**Reply by ervinpopescu** [reply]
`lazy.spawn` also has a `shell` parameter which can be set to `True`.

**Reply by bodograumann** [reply]
Of course, that makes so much sense.
I just wish this fact would be more easily discoverable from the docs.

The main point where I read about `lazy.spawn` was here: https://docs.qtile.org/en/latest/manual/config/lazy.html#general-functions
That page makes some references to the [commands api](https://docs.qtile.org/en/latest/manual/commands/api/index.html), but links to an index page with 8 subpages.
Now that I already know what to look for exactly, I also see

> To access commands on this object via the command graph, use one of the following options:

and I can find `spawn` in the [Qtile root object](https://docs.qtile.org/en/latest/manual/commands/api/root.html) page.
Even the full API is visible [further down](https://docs.qtile.org/en/latest/manual/commands/api/root.html#libqtile.core.manager.Qtile.spawn).

Maybe there could be some direct link from each lazy command to the docs of the actual method?

**Reply by bodograumann** [reply]
Huh, turns out `lazy.spawn("LD_PRELOAD=/usr/lib/libwcwidth-icons.so urxvt")` does not actually work. It just logs

> 2024-02-20 10:37:53,608 ERROR libqtile manager.py:spawn():L1255 couldn't find `LD_PRELOAD=/usr/lib/libwcwidth-icons.so`

I'll create a bugticket.

**Reply by elParaguayo** [reply]
Did you try `lazy.spawn("LD_PRELOAD=/usr/lib/libwcwidth-icons.so urxvt", shell=True)`

Edit, sorry, just saw the issue you posted.

**Reply by bodograumann** [reply]
Ah yes, that is what I meant :-) Simply forgot the `shell = true` in the comment above.
